File: src/entity.py
import pygame
from object import *
from sharedVars import *
#/* Enum - entity_directions */
ENTITY_DIR_NONE = 0
ENTITY_DIR_UP = 1
ENTITY_DIR_DOWN = 2
ENTITY_DIR_LEFT = 3
ENTITY_DIR_RIGHT = 4
#/* Enum End */
class entity:
    def __init__(
        self,
        red, green, blue,
        direction,
        width, height,
        x, y,
        speed,
        visible
        ):
        self.direction = direction
        self.speed = speed
        self.object = objectClass(
            red, green, blue,
            width, height,
            x, y,
            visible
            )
        return
    def draw(
        self,
        screen
        ):
        #/* Why doesn't Python have switches? */
        if (self.direction == ENTITY_DIR_UP):
            self.object.position[1] -= self.speed
        elif (self.direction == ENTITY_DIR_DOWN):
            self.object.position[1] += self.speed
        elif (self.direction == ENTITY_DIR_LEFT):
            self.object.position[0] -= self.speed
        elif (self.direction == ENTITY_DIR_RIGHT):
            self.object.position[0] += self.speed
        return self.object.draw(screen)

File: src/entity.c
#include "entity.h"

void entity_init(
	entity_t *entity,
	unsigned char red,
	unsigned char green,
	unsigned char blue,
	enum entity_directions direction,
	unsigned short width,
	unsigned short height,
	signed short x,
	signed short y,
	signed short speed,
	char visible
) {
	entity->direction = direction;
	entity->speed = speed;
	object_init(
		&entity->object,
		red, green, blue,
		width, height,
		x, y,
		visible
	);
	return;
}

void entity_draw(entity_t *entity) {
	if (entity->object.visible) {
		if (entity->direction != ENTITY_DIR_NONE) {
			entity->object.position[
				entity->direction >= ENTITY_DIR_LEFT ? 0 : 1
			] += (
				(int)entity->direction % 2
			) == 0 ? -entity->speed : entity->speed;
		}
		object_draw(&entity->object);
	}
	return;
}